src/main/java/formflow/library/FileController.java

```diff
@@ -50,6 +62,18 @@ public UploadController(
     this.messageSource = messageSource;
   }
 
+  /**
+   * File upload endpoint.
+   *
+   * @param file         A MultipartFile file
+   * @param flow         The current flow name, not null
+   * @param inputName    The current inputName
+   * @param thumbDataUrl The thumbnail URL generated from the upload
+   * @param httpSession  The current httpSession
+   * @return ON SUCCESS: A ResponseEntity with a status of (HttpStatus.OK), a * contentType of (MediaType.TEXT_PLAIN) and a
+   * body(newFileId.toString()).
```

Comment:
For the `@return` - take out the "ON SUCCESS" (unless this is a java standard??).  Also, you can just say "ResponseEntity with a body containing the id of the file, if successfully uploaded, else an error code will be returned." here, no one will understand the "newFileId.toString()" -> that won't make sense to them unless they read the code, which they shouldn't have to. 

These javadocs should provide enough abstracted  information (word description, not code) about the function so that they don't have to read the code.

Reply:
Here's an example of a rendered Javadoc and how these javadocs might look once we render them - this does not include the code, so referencing variables in the code is not helpful at all: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#replace(java.lang.CharSequence,java.lang.CharSequence)